lms/djangoapps/course_api/api.py

```diff
@@ -266,3 +269,123 @@ def get_course_run_url(request, course_id):
+        - Paginating
+            get_course_members(course_key, page=2)
+        - Paginating with custom limit
+            get_course_members(course_key, limit=50)
+
+    Arguments:
+        course_key (CourseKey): the CourseKey for the course
+        include_students: wether or not to include students,
+        access_roles:
+            accepts an array of string course access roles.
+            If None provided, it includes all roles.
+        page (int): the page number for pagination
+        limit (int): how many results to include per page
+
+    Returns:
+        dict: A dictionary with paginated result with following format -
+            {
+                'count': total number of user instances matching specified filter,
+                'num_pages': total number of pages,
+                'current_page': current page number,
+                'result': [
+                    {
+                        'id': user id,
+                        'email': user's email,
+                        'username': user's username,
+                        'profile': {
+                            'name': user's name,
+                            'profile_image': user's profile image,
+                            .... other profile information
+                        },
+                        'enrollments': [
+                            {
+                                'mode': enrollment mode
+                            }
+                        ],
+                        'course_access_roles': [
+                            {
+                                'role': user's role in course
+                            }
+                        ]
+                    }
+                ]
+            }
+    """
+    queryset = User.objects.filter(is_active=True).order_by('id')
+
+    # if access_roles not given, assign all registered access roles
+    if access_roles is None:
+        access_roles = list(REGISTERED_ACCESS_ROLES.keys())
+
+    # conditions for filtering based on CourseAccessRole
+    access_role_qs = Q(
+        courseaccessrole__course_id=course_key,
+        courseaccessrole__role__in=access_roles
+    )
+
+    # conditions for filtering based on CourseEnrollment
+    students_qs = Q(
+        courseenrollment__course_id=course_key,
+        courseenrollment__is_active=True
+    )
+
+    if include_students:
+        queryset = queryset.filter(access_role_qs | students_qs)
+    else:
+        queryset = queryset.filter(access_role_qs)
+
+    # prevent duplicates
+    queryset = queryset.distinct()
+
+    # prefetch UserProfile
+    queryset = queryset.prefetch_related('profile')
+
+    # prefetch CourseAccessRole items related to the course and given roles
+    queryset = queryset.prefetch_related(Prefetch(
+        'courseaccessrole_set',
+        CourseAccessRole.objects.filter(
+            course_id=course_key, role__in=access_roles)
```

Comment:
Minor minor nit: line length in edx repos is 120, so no need to wrap this line awkwardly.  Make sure your editor reads editorconfig files, they're great :)

Reply:
Fixed-line break.